proxmoxtf/resource_virtual_environment_vm.go

```diff
@@ -2488,7 +2488,9 @@ func resourceVirtualEnvironmentVMGetDiskDeviceObjects(
 		diskDevice.Size = &sizeString
 		diskDevice.SizeInt = &size
 		diskDevice.IOThread = &ioThread
-		diskDevice.SSD = &ssd
+        if strings.Contains(diskInterface, "scsi") {
```

Comment:
From the PVE docs:
> Note that SSD emulation is not supported on VirtIO Block drives.

so I think this should be reflected in the condition then, i.e. `if !strings.HasPrefix(diskInterface, 'virtio') {...}`.

Also linter doesn't like cobbling `if` statements with unrelated assignments, so you may want to move this `if` block down a little bit, after the line 2494